### PR TITLE
Library reloading on SIGHUP

### LINKED ISSUE #243: kiwix-serve should reload automatically content (library/files)
If the files open (library/ZIM files) change, then kiwix-serve should automatically detect it and reload them (or load new files and stop providing removed files). All of this should happen without service interruption. The idea here is to allow to update the library without having here to restart the whole service and create a service interruption. The reload (and not restart) should ideally be smart and do modifications in the internal library only where this is necessary.

### PR DESCRIPTION
**Pull request by veloman-yunkan**
Fixes #243 and #502

Depends on kiwix/libkiwix#636

Library reloading updates metadata of existing books, adds new books and/or removes books that are no longer present in the library XML file.

### CODE REVIEW

File: src/server/kiwix-serve.cpp [hunk 1]
```diff
@@ -123,9 +128,20 @@ void setup_sighandlers()
...
 }
 #endif
 
+void reloadLibrary(kiwix::Manager& mgr, const std::vector<std::string>& paths)
+{
+    try {
+      mgr.reload(paths);
+    } catch ( const std::runtime_error& err ) {
+      cerr << err.what() << endl;
+      exit(1);
```
[the remark this answers is not in the document]
Comment (veloman-yunkan) [reply src/server/kiwix-serve.cpp]: Now we don't

File: src/server/kiwix-serve.cpp [hunk 2]
```diff
@@ -361,6 +352,11 @@ int main(int argc, char** argv)
     }
 
     kiwix::sleep(1000);
+    if ( library_reload_requested && !libraryPaths.empty() ) {
+      reloadLibrary(manager, libraryPaths);
+      nameMapper.update();
+      library_reload_requested = false;
+    }
```
[the remark this answers is not in the document]
Comment (veloman-yunkan) [reply src/server/kiwix-serve.cpp:368]: Done

File: src/server/kiwix-serve.cpp [hunk 3]
```diff
@@ -123,9 +128,27 @@ void setup_sighandlers()
...
     set_signal_handler(SIGTERM, &handle_sigterm);
     set_signal_handler(SIGINT,  &handle_sigterm);
+    set_signal_handler(SIGHUP,  &handle_sighup);
 }
 #endif
 
+bool reloadLibrary(kiwix::Manager& mgr, const std::vector<std::string>& paths)
+{
+    try {
+      std::cerr << "Loading the library from the following files:\n";
```
Comment (mgautierfr) [inline src/server/kiwix-serve.cpp:138]: Do we want to print this on `cerr` ?
I would say that it should go to `cout`.
But I don't know if there are some special rules for a log coming from a daemon reloading the config.

Comment (veloman-yunkan) [reply src/server/kiwix-serve.cpp:138]: `stderr` is used for diagnostic output if the program sends useful output to `stdout`. `kiwix-serve` doesn't produce any such output so we can print this message to `cout` instead. I have absolutely no preference here.

File: src/server/kiwix-serve.cpp [hunk 4]
```diff
@@ -71,7 +71,6 @@ void usage()
             << "\t-V, --version\t\tprint software version" << std::endl
             << "\t-z, --nodatealiases\tcreate URL aliases for each content by removing the date" << std::endl
             << "\t-c, --customIndex\tadd path to custom index.html for welcome page" << std::endl
-            << "\t--donottrustlibrary\tRead the metadata from the zim file instead of trusting the library." << std::endl
```
[the remark this answers is not in the document]
Comment (veloman-yunkan) [reply src/server/kiwix-serve.cpp:74]: @kelson42 So is it fine to remove the `--donottrustlibrary` flag?

[the remark this answers is not in the document]
Comment (kelson42) [reply src/server/kiwix-serve.cpp:74]: I have created #502 to remove --donottrustlibrary. We will discuss this separatly.

File: src/server/kiwix-serve.cpp (hunk 3 above)
Comment (mgautierfr) [reply src/server/kiwix-serve.cpp:138]: Please move everything to `cout` then.

Comment (veloman-yunkan) [reply src/server/kiwix-serve.cpp]: Done (but the output printed in case of an error still goes to `stderr`).

**Comment by mgautierfr** [comment]
We are good. Please rebase-fixup before we merge.

**Pull request merged**